### PR TITLE
REF Extract code to build pcp_supporter_text and enable translation

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Extract code to build the pcp_supporter_text which is duplicated in two places and only translated in one of them.

Before
----------------------------------------
Code duplication, missing translation.

After
----------------------------------------
Single function in BAO object to generate pcp_supporter_text. All strings translateable.

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by mattwire** [comment]
test this please

File: CRM/Contribute/Form/Contribution/Main.php [hunk 1]
```diff
@@ -439,12 +439,7 @@ public function buildQuickForm() {
     }
     if ($this->_pcpId && empty($this->_ccid)) {
       if ($pcpSupporter = CRM_PCP_BAO_PCP::displayName($this->_pcpId)) {
-        $pcp_supporter_text = ts('This contribution is being made thanks to the effort of <strong>%1</strong>, who supports our campaign.', [1 => $pcpSupporter]);
-        // Only tell people that can also create a PCP if the contribution page has a non-empty value in the "Create Personal Campaign Page link" field.
-        $text = CRM_PCP_BAO_PCP::getPcpBlockStatus($this->_id, 'contribute');
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution/Main.php]: You've actually made a change here - you are passing in $this->_pcpId but then using it in place of $this->_id

File: CRM/Contribute/Form/Contribution/Main.php [hunk 2]
```diff
@@ -439,12 +439,7 @@ public function buildQuickForm() {
     }
     if ($this->_pcpId && empty($this->_ccid)) {
       if ($pcpSupporter = CRM_PCP_BAO_PCP::displayName($this->_pcpId)) {
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution/Main.php]: Variable no longer used?

Comment (mattwire) [reply CRM/Contribute/Form/Contribution/Main.php]: Removed

File: CRM/Contribute/Form/Contribution/Main.php (hunk 1 above)
Comment (mattwire) [reply CRM/Contribute/Form/Contribution/Main.php]: @eileenmcnaughton Thanks for spotting - I've now added the contributionPageID as a parameter so nothing is being changed anymore.

**Comment by mattwire** [comment]
@eileenmcnaughton Thanks for your review - I think I've addressed the issues.

**Comment by eileenmcnaughton** [comment]
I worked through this & tested it on the contribution page & it looked fine.

One note is that this
```
 if (CRM_PCP_BAO_PCP::displayName($this->_pcpId)) {
        $pcp_supporter_text = 
        $this->assign('pcpSupporterText', $pcp_supporter_text);
      }

``

Could be 
```
$this->assign('pcpSupporterText', CRM_PCP_BAO_PCP::getPcpSupporterText($this->_pcpId, $this->_id, 'contribute'));
```

the outer if adds a query for no gain

**Comment by mattwire** [comment]
@eileenmcnaughton Thanks for merging. I think I agree with your comments but I just left it more or less as it was as I wasn't sure when you would not have a displayName etc. and did not want to introduce more risk.

**Pull request merged**